distributed/shuffle/arrow.py

```diff
@@ -0,0 +1,62 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import pyarrow as pa
```

Comment:
This PR introduces pyarrow as an optional dependency. We typically deal with optional imports by using a 

```
try:
...
except ImportError:
    # raise useful warning
```